src/Imagine/Cache/Resolver/PsrCacheResolver.php

```diff
@@ -202,22 +173,19 @@ private function generateIndexKey($cacheKey)
         ]);
     }
 
-    /**
-     * @param string $cacheKeyPart
-     *
-     * @return string
-     */
-    private function sanitizeCacheKeyPart($cacheKeyPart)
+    private function sanitizeCacheKeyPart(?string $cacheKeyPart): string
     {
+        if (null === $cacheKeyPart) {
+            return '';
+        }
+
```

Comment:
this is because:

https://github.com/liip/LiipImagineBundle/pull/1422/files#diff-3ffe7291fc61244af59d9d29370815bc777fd8ef61d1b5035851ee7fd9c82f00R100

```php
$this->removePathAndFilter(null, $filter);
```

So the `string` passed can be `null`

Reply:
does the place calling it with null end up with a safe cache key, or do we have to fix something on that side? it feels a bit strange, but maybe its ok

Reply:
I've split this method into two (in a new commit):

https://github.com/liip/LiipImagineBundle/blob/1fb3dd3dd6f99a9fdfc906189f59cbb821e4a215/src/Imagine/Cache/Resolver/PsrCacheResolver.php#L152-L182

- `removePathAndFilter` now receives a `string $path` and `string $filter`
- `removeFilter` that only receives a `string $filter`